src/main/groovy/com/twcable/grabbit/GrabbitConfiguration.groovy

```diff
@@ -117,6 +119,20 @@ class GrabbitConfiguration {
         )
     }
 
+    /**
+     * Sorts the {@code pathConfigurations} on the basis of path depth
```

Comment:
"Sorts the .... on the basis of path depth". 

That's not really what we are doing below. We are sorting paths within the collection lexicographically.

What about an obscure case where we have two paths, "/foo/." - for creating just the root node, and "/foo/$" for creating a node named "$", under "/foo".  "/foo/$" would come first lexicographically. 

I think we need to strip the leading node (or '.'), and forward slash; then compare by counting the number of forward slashes within paths.